tapestry-tool/h5p-kaltura-integration
# 1234 - Add Kaltura integration to H5P videos

## Changes
**Fixes**:
* Fixes errors during npm install by upgrading some packages.
* Fixes code that observes changes in the H5P editor iframe. It should now get elements correctly and observe them without errors.
* Removes references to "UBC Kaltura" from some frontend texts, since users can now supply their own Kaltura account information.
* Removes and ignores files in the `/assets/dist` folder, since these are regenerated with each build.
* Fixes some typos.
* Lints files.

**Changes**:
* Adds back the option to add an H5P video via upload.
* Detects Kaltura configuration variables through site options, if available, or otherwise through wp-config.php, to integrate with Tapestry Tool (see https://github.com/tapestry-tool/tapestry-wp/pull/1248 for more details).
* Displays an error notice when Kaltura configuration variables are not detected.
* Adds option to upload a video directly to Kaltura. When upload completes, the Kaltura ID and video URL fields are automatically populated. Uploaded videos are categorized under `Tapestry>{site URL}>{date}>H5P` on Kaltura.

## Screenshot
Adding Kaltura video from Kaltura ID:

https://user-images.githubusercontent.com/44929104/187593671-bd6bd777-bcb7-435c-ac3d-2f3002d2b44a.mp4

Notice if Kaltura is not available:
![image](https://user-images.githubusercontent.com/44929104/187759478-d1358aa6-039b-4955-a56c-81e5ec79672d.png)

## Issue Linkage
Associated with https://github.com/tapestry-tool/tapestry-wp/issues/1234

## PR Dependency
Associated with https://github.com/tapestry-tool/tapestry-wp/pull/1237

## Automated Testing
N/A

## Testing
To run the plugin in development mode:
* Ensure your Kaltura configuration variables are set in `wp-config`.
* Ensure you have the H5P plugin installed and run `npm start`. On Windows, you may have to modify the `npm start` command in package.json for this to work, since the commands are currently specific to Unix/Linux.
* Create a new H5P video, and select the Interactive Video content type.
* Under "Add a Video", click the box with a "+". If you see the option "Use UBC Kaltura Video", the plugin's functionality has been added correctly.
